test/testdrive/joins.td

```diff
@@ -239,3 +239,41 @@ Unable to automatically determine a timestamp for your query; this can happen if
 
 ! SELECT * FROM names FULL OUTER JOIN mods_unmat ON 1 = 0;
 Unable to automatically determine a timestamp for your query; this can happen if your query depends on non-materialized sources
+
+# Test joins on implicitly coercible data types
```

Comment:
Is there a reason this needs to be tested in testdrive as opposed to the sqllogictests?

Reply:
Agreed that an SLT here would be best!